notebooks/samples/105 - Regression with DataConversion.ipynb

```diff
@@ -0,0 +1,209 @@
+    "- `TrainRegressor`: [TrainRegressor](http://mmlspark.azureedge.net/docs/pyspark/TrainRegressor.html)\n",
+    "- `ComputePerInstanceStatistics`: [ComputePerInstanceStatistics](http://mmlspark.azureedge.net/docs/pyspark/ComputePerInstanceStatistics.html)\n",
+    "- `DataConversion`: [DataConversion](http://mmlspark.azureedge.net/docs/pyspark/DataConversion.html)\n",
+    "\n",
+    "First, import packages"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "import pandas as pd"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next, import the CSV dataset"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "dataFile = \"On_Time_Performance_2012_9.csv\"\n",
+    "import os, urllib\n",
+    "if not os.path.isfile(dataFile):\n",
+    "    urllib.request.urlretrieve(\"https://mmlspark.azureedge.net/datasets/\"+dataFile, dataFile)\n",
+    "flightDelay = spark.createDataFrame(pd.read_csv(dataFile))\n",
+    "#print some basic info\n",
+    "print(\"records read: \" + str(flightDelay.count()))\n",
+    "print(\"Schema: \")\n",
+    "flightDelay.printSchema()\n",
+    "flightDelay.limit(10).toPandas()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Use the `DataConversion` API to convert the columns listed to double."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "from mmlspark import DataConversion\n",
+    "flightDelay = DataConversion(col=\"Quarter,Month,DayofMonth,DayOfWeek,OriginAirportID,DestAirportID,CRSDepTime,CRSArrTime\",\n",
+    "                             convertTo=\"double\").transform(flightDelay)\n",
+    "flightDelay.printSchema()\n",
+    "flightDelay.limit(10).toPandas()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Split the datasest into train and test sets."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "train, test = flightDelay.randomSplit([0.75, 0.25])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Train a regressor on the dataset with l-bfgs"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "from mmlspark import TrainRegressor, TrainedRegressorModel\n",
+    "from pyspark.ml.regression import LinearRegression\n",
+    "from pyspark.ml.feature import StringIndexer\n",
+    "\n",
+    "# Convert columns to categorical\n",
+    "catCols = [\"Carrier\", \"DepTimeBlk\", \"ArrTimeBlk\"]\n",
+    "for catCol in catCols:\n",
```

Comment:
Can the DataConversion be used to make the columns into categorical in one step, as opposed to using StringIndexer in a loop?

Reply:
Currently, DataConversion only coverts data types, but does not convert to/from categorical the way that V1 metadata editor did.